sabakan/generator.go

```diff
@@ -234,108 +195,104 @@ func (g *Generator) getWorkerTmpl(role string) nodeTemplate {
 	panic("BUG: instantiating for invalid role: " + role)
 }
 
-// selectMachine selects a healthy unused machine.
-// If there is no such machine, this returns nil.
-func (g *Generator) selectMachineFromUnused(cp bool, numMachinesByRack map[int]int) *Machine {
-	tmpl := g.cpTmpl
-	if !cp {
-		tmpl = g.chooseWorkerTmpl()
-	}
+// SetWaitSeconds set seconds before removing retiring nodes from the cluster.
+func (g *Generator) SetWaitSeconds(secs float64) {
+	g.waitSeconds = secs
+}
 
-	var unused []*Machine
-	if tmpl.Role == "" {
-		unused = g.nextUnused
-	} else {
-		for _, m := range g.nextUnused {
-			if m.Spec.Role == tmpl.Role {
-				unused = append(unused, m)
-			}
+func removeMachine(machines []*Machine, m *Machine) []*Machine {
+	var ret []*Machine
+	for i, mm := range machines {
+		if m == mm {
+			ret = append(machines[:i], machines[i+1:]...)
+			break
 		}
 	}
-	if len(unused) == 0 {
+	return ret
```

Comment:
If `m` is not found in `machines`, this function returns nil.
Is this intended?  If so, please add a comment to warn users.